src/static_canvas.class.js

```diff
@@ -1770,6 +1770,13 @@
       this.overlayImage = null;
       this._iTextInstances = null;
       this.contextContainer = null;
+      // restore canvas size to original size in case retina scaling was applied
+      this.lowerCanvasEl.setAttribute('width', this.width);
+      this.lowerCanvasEl.setAttribute('height', this.height);
+      if (this.upperCanvasEl) {
+        this.upperCanvasEl.setAttribute('width', this.width);
+        this.upperCanvasEl.setAttribute('height', this.height);
+      }
```

Comment:
the upperCanvasElement goes away with Canvas, that is the class that instantiate it.
While lowerCanvasEl gets back into the dom on place of the original one, so resizing it as you did is a good idea.

Reply:
I didn't understand what you meant about upperCanvasElement. 
You mean that it gets removed so there's no point in these code lines?

Reply:
Yes if you have a StaticCanvas you have no `upperCanvasEl`. if you have a Canvas, you call Canvas.dispose() that will first remove the upperCanvasEl, and then call his parent function, this one.
so when this function runs there should be never a upperCanvasEl

Reply:
4c03569